quora-api/src/main/java/com/upgrad/quora/api/Controller/QuestionController.java

```diff
@@ -0,0 +1,142 @@
+
+import com.upgrad.quora.api.model.*;
+import com.upgrad.quora.service.business.QuestionBusinessService;
+import com.upgrad.quora.service.entity.QuestionEntity;
+import com.upgrad.quora.service.entity.UserAuthTokenEntity;
+import com.upgrad.quora.service.exception.AuthorizationFailedException;
+import com.upgrad.quora.service.exception.InvalidQuestionException;
+import com.upgrad.quora.service.exception.UserNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.ZonedDateTime;
+import java.util.*;
+
+
+@RestController
+@RequestMapping("/")
+public class QuestionController {
+
+    @Autowired
+    QuestionBusinessService questionBusinessService;
+
+
+//the method is used to create the question and checks the authorization
+    @RequestMapping(method = RequestMethod.POST, path ="/question/create", consumes = MediaType.APPLICATION_JSON_UTF8_VALUE, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+      public ResponseEntity<QuestionResponse> createQuestion(final QuestionRequest questionRequest, @RequestHeader("authorization") final String authorization) throws AuthorizationFailedException {
+       final  QuestionEntity questionEntity = new QuestionEntity();
+
+        questionEntity.setUuid(UUID.randomUUID().toString());
+        questionEntity.setContent(questionRequest.getContent());
+        questionEntity.setDate(ZonedDateTime.now());
+
+          QuestionEntity  createdQuestionEntity;
+        try{
+            String[] bearerAccessToken = authorization.split("Bearer ");
+            createdQuestionEntity = questionBusinessService.postQuestion(questionEntity, bearerAccessToken[1]);
+        } catch(ArrayIndexOutOfBoundsException are) {
+            createdQuestionEntity = questionBusinessService.postQuestion(questionEntity, authorization);
+        }
+
+        QuestionResponse questionResponse = new QuestionResponse().id(createdQuestionEntity.getUuid()).status("QUESTION CREATED");
+
+        return new ResponseEntity<QuestionResponse>(questionResponse, HttpStatus.CREATED);
+    }
+
+//this methods gets all the question
+    @RequestMapping(method = RequestMethod.GET, path ="/question/all", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<List<QuestionDetailsResponse>> getAllQuestions(@RequestHeader("authorization") final String authorization) throws AuthorizationFailedException{
+        List<QuestionEntity> listOfQuestions = new ArrayList<>();
+        try{
+            String[] bearerAccessToken = authorization.split("Bearer ");
+            listOfQuestions = questionBusinessService.getAllQuestions(bearerAccessToken[1]);
+        }catch(ArrayIndexOutOfBoundsException are) {
+            listOfQuestions = questionBusinessService.getAllQuestions(authorization);
```

Comment:
A thought here. The authorization header needs to be of a standard format of "Bearer <token_value>". If its not in this format, then I feel that the request should not be processed further.